cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -366,12 +388,90 @@ class device_row_comparator {
         rhs_element_index);
     }
 
+    template <typename Element,
+              CUDF_ENABLE_IF(not cudf::is_relationally_comparable<Element, Element>() and
+                             std::is_same_v<Element, cudf::list_view>)>
+    __device__ cuda::std::pair<weak_ordering, int> operator()(size_type lhs_element_index,
+                                                              size_type rhs_element_index)
+    {
+      auto const l_offsets    = _l_dremel_device_view.offsets;
+      auto const r_offsets    = _r_dremel_device_view.offsets;
+      auto l_start            = l_offsets[lhs_element_index];
+      auto l_end              = l_offsets[lhs_element_index + 1];
+      auto r_start            = r_offsets[rhs_element_index];
+      auto r_end              = r_offsets[rhs_element_index + 1];
+      column_device_view lcol = _lhs.slice(lhs_element_index, 1);
+      column_device_view rcol = _rhs.slice(rhs_element_index, 1);
+      while (lcol.type().id() == type_id::LIST) {
+        lcol = detail::lists_column_device_view(lcol).get_sliced_child();
+        rcol = detail::lists_column_device_view(rcol).get_sliced_child();
+      }
+      auto const l_max_def_level = _l_dremel_device_view.max_def_level;
+      auto const l_def_levels    = _l_dremel_device_view.def_levels;
+      auto const r_def_levels    = _r_dremel_device_view.def_levels;
+      auto const l_rep_levels    = _l_dremel_device_view.rep_levels;
+      auto const r_rep_levels    = _r_dremel_device_view.rep_levels;
+      PRINTF("max_def_level: %d\n", l_max_def_level);
+
+      PRINTF("t: %d, lhs_element_index: %d, rhs_element_index: %d\n",
+             threadIdx.x,
+             lhs_element_index,
+             rhs_element_index);
+      PRINTF("t: %d, l_start: %d, l_end: %d, r_start: %d, r_end: %d\n",
+             threadIdx.x,
+             l_start,
+             l_end,
+             r_start,
+             r_end);
+      weak_ordering state{weak_ordering::EQUIVALENT};
+      for (int i = l_start, j = r_start, k = 0; i < l_end and j < r_end; ++i, ++j) {
```

Comment:
Maybe `i, j, k` ↔️ `left_list_index`, `right_list_index`, `list_element_index`? I can't quite tell what these should be named.

Reply:
I went with `left_dremel_index`, `right_dremel_index`, and `element_index`. It's verbose, but I think it's the only really informative path. We are indexing into the dremel data and you need to understand the dremel encoding well enough to recognize that these indexes include "elements" corresponding to empty lists or nulls.